libc/include/CMakeLists.txt

```diff
@@ -17,676 +17,1301 @@ add_header(
     __llvm-libc-common.h
 )
 
-add_gen_header2(
-  ctype
-  YAML_FILE ../libc/newhdrgen/yaml/ctype.yaml
-  DEF_FILE ctype.h.def
-  GEN_HDR ctype.h
-  DEPENDS
-    .llvm_libc_common_h
-)
-
-add_gen_header2(
-  dirent
-  YAML_FILE ../libc/newhdrgen/yaml/dirent.yaml
-  DEF_FILE dirent.h.def
-  GEN_HDR dirent.h
-  DEPENDS
-    .llvm_libc_common_h
-    .llvm-libc-types.ino_t
-    .llvm-libc-types.DIR
-    .llvm-libc-types.struct_dirent
-)
+if (USE_NEW_HEADER_GEN)
```

Comment:
You may want to add the option command that can switch either on and off depending on how the user wants to build for now. Something like option(USE_NEW_HEADER_GEN OFF) before this line and then define a macro such as macro(new_function_name...) if...

Reply:
option is added in libc/CMakesList.txt. Macro has now been added